interface/VideoCommonDefs.h

```diff
@@ -21,30 +21,27 @@
 #include <stdint.h>
 
 #ifdef __cplusplus
-#ifdef ANDROID
-#include <memory>
-#else
-#include <tr1/memory>
-#endif
-namespace YamiMediaCodec{
-#ifdef ANDROID
-    #define SharedPtr std::shared_ptr
-    #define WeakPtr std::weak_ptr
-    #define DynamicPointerCast std::dynamic_pointer_cast
-    #define StaticPointerCast std::static_pointer_cast
-    #define EnableSharedFromThis std::enable_shared_from_this
-#else
-    #define SharedPtr std::tr1::shared_ptr
-    #define WeakPtr std::tr1::weak_ptr
-    #define DynamicPointerCast std::tr1::dynamic_pointer_cast
-    #define StaticPointerCast std::tr1::static_pointer_cast
-    #define EnableSharedFromThis std::tr1::enable_shared_from_this
-#endif
 
-#define SharedFromThis shared_from_this
-
-}
-#endif
+#if __cplusplus > 199711L
```

Comment:
So on GCC that defaults to `-std=c++98` (GCC 4.9.2), I built libyami with default (i.e. TR1 is used).  Then I set `CXXFLAGS=-std=c++11` and compiled libyami-utils (i.e. TR1 is not used).  Everything seemed to work fine.

Conversely, I did similar on GCC where default is -std=C++11 (GCC 6.1.1)... compiled libyami with `CXXFLAGS=-std=c++98` and libyami-utils with default.  No issues there either.

I ran yamidecode with h264 and mjpeg streams for both cases.

Is this what your concern was?